### PR TITLE
sci-misc/boinc: Clean up (#640822) and fix xlocale.h bug (#639108)

### PR DESCRIPTION
**Pull request by Yamakuzure**
This PR includes the following fixes and updates:

1) Clean up old ebuilds
As it was discussed in bug 640822, said bug is now closed with cleaning up the old ebuilds.
The newest no longer depends on gtk2 based wxWidgets, and remains.

Closes: https://bugs.gentoo.org/640822

2) Remove xlocale.h usage, fixes bug 639108
>=sys-libs/glibc-2.26 dropped xlocale.h support.

From https://sourceware.org/glibc/wiki/Release/2.26#Removal_of_.27xlocale.h.27
 > The nonstandard header xlocale.h has been removed in this release. It was
 > never intended to be included directly by programs other than glibc itself,
 > and it was a strict subset of the standard header locale.h.

This commit removes the inclusion of xlocale.h, and substitutes it with locale.h where necessary.
    
Tested-by: Sven Eden <yamakuzure@gmx.net>
Closes: https://bugs.gentoo.org/639108

3) Only update the icon cache, if USE="X" is enabled
    
Package-Manager: Portage-2.3.19, Repoman-2.3.6

@jlec @SoapGentoo Sorry that this took so long. I have massive problems "convincing" my damn broadcom wifi chip to work properly. I hope these commits are alright for you.
I did not revision bump the ebuild, because it bears no change/advantage for users who already had boinc-7.8.4 merged successfully.

### CODE REVIEW

File: sci-misc/boinc/boinc-7.8.4.ebuild [hunk 1]
```diff
@@ -147,7 +147,7 @@ pkg_preinst() {
 }
 
 pkg_postinst() {
-	gnome2_icon_cache_update
+	use X && gnome2_icon_cache_update
```
Comment (leio) [inline sci-misc/boinc/boinc-7.8.4.ebuild]: As already mentioned in #6530, please don't do this conditional to use X. You would have to be sure there are no icons installed then on each bump, etc. Much better to just do what gnome2.eclass does in this case. Put back the savelist in preinst, and then actually use the result in the same way gnome2.eclass does.

File: sci-misc/boinc/boinc-7.8.4.ebuild [hunk 2]
```diff
@@ -185,5 +185,5 @@ pkg_postinst() {
 }
 
 pkg_postrm() {
-	gnome2_icon_cache_update
+	use X && gnome2_icon_cache_update
```
Comment (leio) [inline sci-misc/boinc/boinc-7.8.4.ebuild]: same as in postinst

**Comment by Yamakuzure** [comment]
@leio : I am sorry, it seems that I have misunderstood that part of the (fragmented) conversation. I have changed the commit, so that it updates the icon cache like the gnome2.eclass would do it. Hopefully this is correct now. I do not have any experience with the gnome stuff...

**Comment by leio** [comment]
Well, I changed it from what it was after someones PR got merged (pointless savelist call without using it like you do now), as it's not exactly a terribly expensive call anyhow.

**Comment by leio** [comment]
Merged, thanks

**Pull request closed without merging**